### PR TITLE
[refactor] [fsdp] Modify FSDP API param name to better reflect functionality

### PR DESCRIPTION
**Pull request by anj-s**
## What does this PR do?
Renames the input param `cpu_offload` to `move_params_to_cpu` to better reflect functionality. 

I added a note that we will be deprecating `cpu_offload` in an upcoming release.

## Before submitting

- [x] Did you have fun?
  - Make sure you had fun coding 🙃
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [x] N/A
- [ ] Did you make sure to update the docs?
  - [x] N/A
- [ ] Did you write any new necessary tests?
  - [x] N/A
- [x] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/master/CHANGELOG.md)? (if needed)
  - [ ] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py [hunk 1]
```diff
@@ -348,6 +353,10 @@ def set_gradient_divide_factors(self, pre: float, post: float, recursive: bool)
         self.gradient_predivide_factor = pre
         self.gradient_postdivide_factor = post
 
+    @property
```
Comment (min-xu-ai) [inline fairscale/nn/data_parallel/fully_sharded_data_parallel.py:356]: how about we move all deprecated code to the end of the class?

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py [hunk 2]
```diff
@@ -230,6 +234,7 @@ def __init__(
         clear_autocast_cache: bool = False,
         force_input_to_fp32: bool = False,
         verbose: bool = False,
+        cpu_offload: bool = False,
```
Comment (min-xu-ai) [inline fairscale/nn/data_parallel/fully_sharded_data_parallel.py:237]: perhaps add a comment line above to separate, like:
```suggestion
        # deprecated args below.
        cpu_offload: bool = False,
```

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py [hunk 3]
```diff
@@ -209,6 +209,10 @@ class FullyShardedDataParallel(nn.Module):
         verbose (bool):
             Set this to ``True`` to turn on verbose output for model's string representation.
             Default: False
+        cpu_offload (bool, Optional):
```
Comment (min-xu-ai) [inline fairscale/nn/data_parallel/fully_sharded_data_parallel.py:212]: how about
```suggestion
        # deprecated args below.
        cpu_offload (bool, Optional):
```

**Review by min-xu-ai — COMMENTED** [review]
some quick suggestions. :-)

**Review by min-xu-ai — COMMENTED** [review]
Do you mind I take over this because I am thinking of arrange the params in a more user friendly way.

**Comment by anj-s** [comment]
> Do you mind I take over this because I am thinking of arrange the params in a more user friendly way.

if there are no further comments other these ones I can submit right now.

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py (hunk 3 above)
Comment (anj-s) [reply fairscale/nn/data_parallel/fully_sharded_data_parallel.py:212]: I think I have note in the description so I don't think we need this comment. Feel free to change that in your upcoming refactor if you feel strongly.

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py (hunk 2 above)
Comment (anj-s) [reply fairscale/nn/data_parallel/fully_sharded_data_parallel.py:237]: Same comment as above.

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py (hunk 1 above)
Comment (anj-s) [reply fairscale/nn/data_parallel/fully_sharded_data_parallel.py]: Done.

**Review by min-xu-ai — APPROVED** [review]
We talked, this can go in first.

**Pull request merged**